go-controller/pkg/ovn/master.go

```diff
@@ -793,7 +792,17 @@ func (oc *DefaultNetworkController) addUpdateLocalNodeEvent(node *kapi.Node, nSy
 			errs = append(errs, fmt.Errorf("failed to set up hybrid overlay logical switch port for %s: %v", node.Name, err))
 		}
 	} else {
-		// the node needs to cleanup Hybrid overlay annotations if it has them and hybrid overlay is not enabled
+		// the node needs to cleanup Hybrid overlay annotations LogicalRouterPolicies and Hybrid overlay port
+		// if it has them and hybrid overlay is not enabled
+		portName := util.GetHybridOverlayPortName(node.Name)
+		lsp := nbdb.LogicalSwitchPort{Name: portName}
+		sw := nbdb.LogicalSwitch{Name: node.Name}
+		if err := libovsdbops.DeleteLogicalSwitchPorts(oc.nbClient, &sw, &lsp); err != nil && !errors.Is(err, libovsdbclient.ErrNotFound) {
+			errs = append(errs, fmt.Errorf("failed to delete hybrid overlay switch port for node %s: %v", node.Name, err))
+		}
```

Comment:
You already have `oc.deleteHybridOverlayPort` that does this. And probably `oc.deleteHybridOverlayPort` should also do `oc.removeHybridLRPolicySharedGW` since they both go together always and it is inline with what `oc.handleHybridOverlayPort` does which also setups both things, so it would make sense that `oc.deleteHybridOverlayPort` removes both things.

Reply:
I went to go do this but it seems like this is a little more complicated then it seems. There are additions like https://github.com/ovn-org/ovn-kubernetes/blob/4164dc466468cec538c16bcee14a37517f03f013/go-controller/pkg/ovn/master.go#L906-L911
Where I need to figure out why there are cases like this. I agree that this should happen but I do not know all the ramifications of combining `deleteHybridOverlayPort()` and `removeHybridLRPolicySharedGW()`. Would you settle for an upstream issue/JIRA card for combining the two functions so the BZ can get resolved quicker?

Reply:
@jcaamano would that work for you?

Reply:
Those check additions don't make a lot of sense to me currently.

If the node is being removed, its logical switch is going to be removed and as a consequence, the HO port is also going to be removed regardless of whether `deleteHybridOverlayPort`, which is the only thing it does currently, is called or not.

The only possible purpose of those checks is to avoid attempting to remove a port that does not exist, which might have failed in the past, but does not currently.

So I wouldn't be to concerned about those checks, we can live without them. Otherwise it would make sense to do them as well in the current piece of code that you are adding now.

Reply:
combined functions